File: service-gateway/src/main/java/com/contractlens/service/db/postgres/dao/ContractLensAiIntent.java
package com.contractlens.service.db.postgres.dao;


import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;


@Getter
@Setter
@Entity
@Table(
        name = "contractlens_ai_intent",
        uniqueConstraints = {
                @UniqueConstraint(
                        name = "uk_ai_intent_code",
                        columnNames = "intent_code"
                )
        }
)
public class ContractLensAiIntent extends  BaseAuditEntity{

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(
            name = "intent_code",
            nullable = false,
            length = 100
    )
    private String intentCode;

    @Column(
            name = "description",
            columnDefinition = "TEXT"
    )
    private String description;

    @Column(
            name = "route",
            length = 255
    )
    private String route;

    @Column(
            name = "priority",
            nullable = false
    )
    private Integer priority = 0;

    @Column(
            name = "is_active",
            nullable = false
    )
    private Boolean isActive = true;

    @Column(
            name = "version",
            nullable = false
    )
    private Integer version;

}

File: service-ai-newtown/db/postgres/models/contractlens_ai_intent.py
from datetime import datetime

from sqlalchemy import Boolean, DateTime, Integer, String, Text
from sqlalchemy.orm import Mapped, mapped_column

from db.postgres.client import Base


class ContractLensAiIntent(Base):

    __tablename__ = "contractlens_ai_intent"

    id: Mapped[int] = mapped_column(
        Integer,
        primary_key=True,
        autoincrement=True
    )

    intent_code: Mapped[str] = mapped_column(
        String(100),
        nullable=False
    )

    route: Mapped[str] = mapped_column(
        String(255),
        nullable=False
    )

    description: Mapped[str] = mapped_column(
        Text,
        nullable=False
    )

    version: Mapped[int] = mapped_column(
        Integer,
        nullable=False
    )

    priority: Mapped[int] = mapped_column(
        Integer,
        nullable=False
    )

    is_active: Mapped[bool] = mapped_column(
        Boolean,
        nullable=False
    )

    created_by: Mapped[str | None] = mapped_column(
        String(100),
        nullable=True
    )

    updated_by: Mapped[str | None] = mapped_column(
        String(100),
        nullable=True
    )

    created_at: Mapped[datetime] = mapped_column(
        DateTime,
        nullable=False
    )

    updated_at: Mapped[datetime] = mapped_column(
        DateTime,
        nullable=False
    )